apps/admin_audit/lib/AppInfo/Application.php

```diff
@@ -194,59 +201,81 @@ private function fileHooks(IAuditLogger $logger,
 		$fileActions = new Files($logger);
 		$eventDispatcher->addListener(
 			BeforePreviewFetchedEvent::class,
-			function (BeforePreviewFetchedEvent $event) use ($fileActions) {
-				$file = $event->getNode();
-				$fileActions->preview([
-					'path' => mb_substr($file->getInternalPath(), 5),
-					'width' => $event->getWidth(),
-					'height' => $event->getHeight(),
-					'crop' => $event->isCrop(),
-					'mode' => $event->getMode()
+			function (BeforePreviewFetchedEvent $event) use ($logger) {
+				$fileActions = new Files($logger);
+				$fileActions->preview($event);
+			}
+		);
+
+		/*
+		$eventDispatcher->addListener(
+			NodeRenamedEvent::class,
+			function (NodeRenamedEvent $event) use ($fileActions) {
+				$source = $event->getSource();
+				$target = $event->getTarget();
+				$fileActions->rename([
+					'oldpath' => mb_substr($source->getInternalPath(), 5),
+					'newpath' => mb_substr($target->getInternalPath(), 5),
+					'oldid' => $source->getId(),
+					'newid' => $target->getId()
 				]);
```

Comment:
At that point, `$source->getInternalPath()` and `$source->getId()` will throw an error as the source is a `NonExistingFile`.

```suggestion
				$fileActions->rename([
					'oldpath' => mb_substr($source->getPath(), 5),
					'newpath' => mb_substr($target->getPath(), 5),
					'newid' => $target->getId(),
				]);
```

Reply:
@artonge Thank you :). Refactored the rename event as suggested.